OscLib/OscCalcNuFast.h

```diff
@@ -0,0 +1,94 @@
+                         protected analytic::ProbCache<double, T>,
+                         protected analytic::ProbCache<Eigen::ArrayXd, Eigen::ArrayX<T>> {
+  public:
+    _OscCalcNuFast(void);
+    virtual ~_OscCalcNuFast(void);
+    
+    virtual _OscCalcNuFast<T>* Copy(void) const override;
+    
+    // Define setter functions for oscillation parameters.
+    virtual void SetL     (double L       ) override { this->fIsDirty = true; this->fL = L; }
+    virtual void SetRho   (double rho     ) override { this->fIsDirty = true; this->fRho = rho; }
+    virtual void SetDmsq21(const T& dmsq21) override { this->fIsDirty = true; this->fDmsq21 = dmsq21; }
+    virtual void SetDmsq32(const T& dmsq32) override { this->fIsDirty = true; this->fDmsq32 = dmsq32; }
+    virtual void SetTh12  (const T& th12  ) override;
+    virtual void SetTh13  (const T& th13  ) override;
+    virtual void SetTh23  (const T& th23  ) override;
+    virtual void SetdCP   (const T& dCP   ) override;
+    /// Electron density in matter used by NuFast (default = 0.5).
+    virtual void SetYe    (double Ye      ) { this->fIsDirty = true; this->fYe = Ye; }
+    /// Number of Newton steps to improve eigen{value}{vector} convergence (default = 0 seems fine, use 1 or 2 if you need super accuracy).
+    virtual void SetNNewton(double nNewton) { this->fIsDirty = true; this->fNNewton = nNewton; }
+    
+    // We don't need new getter functions except for Ye and number of Newton corrections (specific to NuFast).
+    virtual double GetYe(void) const { return this->fYe; }
+    virtual double GetNNewton(void) const { return this->fNNewton; }
+    
+    /// Implement the NuFast algorithm.
+    virtual T P(int from, int to, double E) override;
+    virtual Eigen::ArrayX<T> P(int from, int to, const std::vector<double>& E) override;
+    virtual Eigen::ArrayX<T> P(int from, int to, const Eigen::ArrayXd& E) override;
+    
+    /// Specialized printer function.
+    virtual void Print(const std::string& prefix = "") const override;
+    
+    /// Implementation of params hash.
+    virtual TMD5* GetParamsHash(void) const override {
+      return _IOscCalcAdjustable<T>::GetParamsHashDefault("NuFast");
+    }
+  protected:
+    void ClearProbCaches(void) {
+      analytic::ProbCache<double, T>::clear();
+      analytic::ProbCache<Eigen::ArrayXd, Eigen::ArrayX<T>>::clear();
+    }
+    /// Copy constructor.
+    _OscCalcNuFast(const _OscCalcNuFast<T>& other);
+    /// from and to aren't necessary for computing the matrix but are used for the return (so the cache doesn't need to be searched for the result).
+    template<class VT, class KVT> inline __attribute__((always_inline)) VT RecomputeProbabilityMatrix(int from, int to, KVT E);
+    // Master probability function called with some combination of primitives, stan::math::vars, and Eigen objects.
+    template<class VT, class KVT> inline __attribute__((always_inline)) VT _P(int from, int to, KVT E);
```

Comment:
That's a good point. I didn't think very hard about aggressively inlining these functions because `OscCalcAnalytic` does similar things.
I didn't make any testing with and without inlining, but I highly doubt it's actually helpful in any measurable way. I want to remove the inlining and assume that it makes no change to runtime.